src/modules/Verifications/containers/ReusableList/useReusableList.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  deleteVerificationAdaptor,
+  getReusableVerificationsAdaptor,
+  ReusableVerification,
+  ReusableVerificationsRes,
+  SingleUseVerificationsRes,
+  Verification,
+  VerificationsRes,
+} from 'src/core/adaptors';
+
+export const useReusableList = (
+  list: ReusableVerification[],
+  setList: (val: ReusableVerification[]) => void,
+  totalItems: number,
+  setOpenModal: (val: { name: 'delete' | 'copy' | 'history'; open: boolean }) => void,
+  selectedId?: string,
+) => {
+  const { t: translate } = useTranslation();
+
+  const [page, setPage] = useState(1);
+  const PER_PAGE = 10;
+  const [total, setTotal] = useState(Math.ceil(totalItems / PER_PAGE));
+
+  const fetchMore = async () => {
+    const res = await getReusableVerificationsAdaptor(page, PER_PAGE);
+    if (!res.data) return;
+    setList(res.data.items);
+    setTotal(Math.ceil(res.data.totalCount / PER_PAGE));
+  };
+
+  useEffect(() => {
```

Comment:
why do you always use `useEffect` for fetching the list with changing pages?
we already have the implemented logic without `useEffect` in other lists (credential list, schema list, ...)